File: app/src/main/java/com/github2136/android_utils/load_more/ListActivity.java
package com.github2136.android_utils.load_more;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

import com.github2136.android_utils.R;
import com.github2136.base.BaseRecyclerAdapter;

import java.util.ArrayList;
import java.util.List;

public class ListActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list);
        RecyclerView rvContent = (RecyclerView) findViewById(R.id.rv_content);
        List<String> s = new ArrayList<>();
        s.add("asdf1");
        s.add("asdf2");
        s.add("asdf3");
        s.add("asdf4");
        s.add("asdf5");
        s.add("asdf6");
        s.add("asdf7");
        s.add("asdf8");
        s.add("asdf9");
        s.add("asdf10");
        ListAdapter adapter = new ListAdapter(this, s);
        adapter.setOnItemClickListener(rvContent, new BaseRecyclerAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(BaseRecyclerAdapter adapter, int position) {
                Toast.makeText(ListActivity.this, "" + position, Toast.LENGTH_SHORT).show();
            }
        });
        adapter.setOnItemLongClickListener(rvContent, new BaseRecyclerAdapter.OnItemLongClickListener() {
            @Override
            public void onItemClick(BaseRecyclerAdapter adapter, int position) {
                Toast.makeText(ListActivity.this, "" + position, Toast.LENGTH_SHORT).show();
            }
        });
        rvContent.setAdapter(adapter);

    }
}

File: app/src/main/java/com/github2136/android_utils/load_more/ListActivity.kt
package com.github2136.android_utils.load_more

import android.graphics.Color
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.github2136.android_utils.R
import com.github2136.base.divider.Divider
import kotlinx.android.synthetic.main.activity_list.*
import java.util.*

/**
 * Created by yb on 2018/11/1.
 */
class ListActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list)
        val s = ArrayList<String>()
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文1")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("中文2")
        s.add("cccccc")
        s.add("cccccc")
        s.add("cccccc")
        s.add("cccccc")
        s.add("ddddd")
        s.add("ddddd")
        s.add("ddddd")
        s.add("ddddd")
        s.add("eeeeee")
        s.add("eeeeee")
        s.add("eeeeee")
        s.add("asdf6")
        s.add("asdf7")
        s.add("asdf8")
        s.add("asdf9")
        s.add("asdf10")
        val adapter = ListAdapter(s)
        rv_content.adapter = adapter
        val d = Divider(this)
        d.backgroundColor = Color.parseColor("#987654")
        d.lineColorTop = Color.parseColor("#ff0000")
        d.lineHeightTop = 0

        d.lineColorBottom = Color.parseColor("#ffff00")
        d.lineHeightBottom = 0

        d.sticky = false
        d.showItemDivider = false
        rv_content.addItemDecoration(d)
        adapter.setOnItemClickListener { position -> Toast.makeText(this@ListActivity, "" + position, Toast.LENGTH_SHORT).show() }
        adapter.setOnItemLongClickListener { position -> Toast.makeText(this@ListActivity, "" + position, Toast.LENGTH_SHORT).show() }
    }
}